test/gurka/test_instructions_internal_intersection.cc

```diff
@@ -0,0 +1,117 @@
+#include "gurka.h"
+#include <gtest/gtest.h>
+
+#if !defined(VALHALLA_SOURCE_DIR)
+#define VALHALLA_SOURCE_DIR
+#endif
+
+using namespace valhalla;
+
+class InstructionsInternalIntersection : public ::testing::Test {
+protected:
+  static gurka::map map;
+
+  static void SetUpTestSuite() {
+    constexpr double gridsize_metres = 10;
+
+    const std::string ascii_map = R"(
```

Comment:
Do we want to add some degenerate cases like start and/or ending on an internal intersection, eg `A->F`, `B->G` etc?

Reply:
sure - can add some more

Reply:
@mandeepsandhu I added test cases